WcaOnRails/vendor/assets/javascripts/jquery.are-you-sure.js

```diff
@@ -68,13 +68,27 @@
       $field.data('ays-orig', getValue($field));
     };
 
-    var checkForm = function(evt) {
+    // Oftentimes, forms have some javascript that runs immediately when fields change.
+    // It makes sense to wait for those events to run (and possibly dirty or clean the form)
+    // before we check to see if the form is dirty. -JFLY
+    var checkForm = function(evt) {//JFLY
+      var that = this;
+      setTimeout(function() {//JFLY
+        checkFormNow.call(that, evt);//JFLY
+      }, 0);//JFLY
+    };//JFLY
+    var checkFormNow = function(evt) {
 
       var isFieldDirty = function($field) {
         var origValue = $field.data('ays-orig');
         if (undefined === origValue) {
           return false;
         }
+        if($field.is(":disabled")) {//JFLY
+          // Disabled fields cannot be dirty, as they won't
+          // submit anything to the server.
```

Comment:
Yeah, I was wondering if that's how it usually works too. It may well be, I don't know.

But I can think of two problems.

1. Checkboxes only get posted if they're checked. So a disabled box is indistinguishable from an unchecked box server side.
1. A more complex controller action that doesn't just write the fields to the DB can be affected in any number of ways.